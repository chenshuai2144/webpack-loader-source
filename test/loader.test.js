import { webpack } from './helpers';

describe('loader', () => {
  it('should works', async () => {
    const stats = await webpack('basic.js');
    const { modules } = stats.toJson();
    const [, demo] = modules;
    expect(demo.source).toMatchSnapshot('separator');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
