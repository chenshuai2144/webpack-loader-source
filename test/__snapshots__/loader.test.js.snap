// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should works: errors 1`] = `Array []`;

exports[`loader should works: separator 1`] = `
"export default function generateErrorTemplate(err) {
  const strToHtml = (str) =>
    (str || '')
      .replace(/&/g, '&')
      .replace(/</g, '<')
      .replace(/>/g, '>')
      .replace(/\\"/g, '')
      .replace(/'/g, \\"'\\")
      .replace(/\\\\[(\\\\d+)m/g, '')
      .replace(/ /g, ' ')
      .replace(/\\\\n/g, '<br />');
  const template = \`
          <!DOCTYPE html> 
          <html>
          <head>
          </head>
          <body>
            <div>
              \${strToHtml(err.toString()) || ''}
            </div>
          </body>
          </html>\`;
  return template;
}
"
`;
